### ISSUE TITLE
Navigation

### ISSUE THREAD

**Issue by emilkovacev** [makeopensource/Authenticator#16]
**Is your feature request related to a problem? Please describe.**
The user needs to be able to navigate between views in the application

### Views
* Home: a.k.a. the application list. A list of all registered applications for 2fa
* Settings: settings for Authenticator app
* New: QR code reader page for registering a new app

### Navigation:
* Home -> Settings (button)
* Settings -> Home (button)
* Home -> New (button)
* New -> Home (button)

**Describe the solution you'd like**
Use React Native's built-in [React Navigation library](https://reactnative.dev/docs/navigation).
The steps to completion should be as follows:

- [ ] Wrap entire app (index.js) in `<NavigationContainer>` tag
```
import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';

const App = () => {
  return (
    <NavigationContainer>
      {/* Rest of your app code */}
    </NavigationContainer>
  );
};

export default App;
```

- [ ] Create a Navigation stack
```
import { createNativeStackNavigator } from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();
```

- [ ] insert navigation links
```
<Button
      title="Go to Jane's profile"
      onPress={() =>
        navigation.navigate('Profile', { name: 'Jane' })
      }
    />
```

- [ ] style as needed

**Comment by mattrrubino** [comment]
I have some ideas for the navigation system. I was thinking we could put a header at the top of the app with navigation buttons and a title indicating what page the user is viewing. How does this look?

![navigation](https://user-images.githubusercontent.com/83928138/147801436-37c7cc4a-502c-47f4-9034-86b293fd9856.PNG)

**Comment by emilkovacev** [comment]
I really like the way it looks, super simple and easy to navigate. All it needs is an "add" button for adding apps and that should cover all the pages we need, right?

**Comment by mattrrubino** [comment]
Yes, I believe so. I ended up opting to put this in the header as well. Here is what all of the pages look like now:

![application-list](https://user-images.githubusercontent.com/83928138/147853892-94c37963-ea60-4698-997c-669bf4b2e3d7.PNG)
![settings](https://user-images.githubusercontent.com/83928138/147853895-6d50ae86-5dc3-43d7-8a60-3cfc71035740.PNG)
![new](https://user-images.githubusercontent.com/83928138/147853896-5eddaca7-cb34-415a-a5cd-d1e528f7c536.PNG)

Also, I did not make the icons. They are from https://feathericons.com/, so if we end up using them, we should make sure to credit it somewhere in the repository. Should I make a PR?
